### DISCUSSION TITLE
Cannot generate sitemap due to unrecognized field

### DISCUSSION THREAD

**Discussion by abdellahrk** [manuxi/SuluNewsBundle#7] [Q&A] [answered]
Hello, I hope this will not be regarded as a personal support but a general feedback. 
Trying to dump the sitemap `bin/websiteconsole sulu:website:dump-sitemap` does not work and brings this error `UnrecognizedField.php line 22:                                                                    
  Unrecognized field: Manuxi\SuluNewsBundle\Entity\News::$published `
How can this be resolved? I see the field in question is a virtual property in `PublishedTranslatableTrait`
Thank you.

**Comment by manuxi** [comment, marked as answer]
Well good catch. Haven't noticed it, this was really an issue.
I fixed it and created a new release, since I'm goind to add some customisation (settings) later.
Afaik you don't have to use the command since it's done with the fe controller already.

**Comment by abdellahrk** [comment]
Hi, here is the next error after updating 
`  Manuxi\SuluNewsBundle\Repository\NewsRepository::countForSitemap(): Argument #1 ($locale) must be of type string, null given, called in /var/www/  
  hey/vendor/manuxi/sulu-news-bundle/src/Sitemap/NewsSitemapProvider.php on line 58`

**Reply by manuxi** [reply]
Which version are you using?

**Reply by abdellahrk** [reply]
Hi @manuxi, everything is fixed. It was due to my misconfiguration.

**Comment by abdellahrk** [comment]
Hello @manuxi, 
Sorry to disturb on this thread again. I see sitemaps are not generated correctly. The sitemap xml file that generates is `news-1.xml` and all what generate in there are just the url of the website. Can you point to me which listener generates this?

**Comment by manuxi** [comment]
Please show your xml. 
Btw. just fixed a method in the NewsSitemapProvider which failed in some circumstances to get the correct locale. Give it a try....

**Reply by abdellahrk** [reply]
Hi @manuxi, this is my sitemap from https://hey.cm/sitemaps/news-1.xml 
<img width="821" alt="news-1.xml" src="https://github.com/manuxi/SuluNewsBundle/assets/17084652/70fe2703-8259-4dbb-9125-62555c302ae8">

**Reply by manuxi** [reply]
This looks not correct. Mine looks like
```
<urlset xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <url>
		<loc>https://[domain]/blog-und-infos/[slug]</loc>
		<lastmod>2023-11-14</lastmod>
	</url>
	[...]
</urlset>
```

**To help, answer the following.**

Do you have any **valid and published** news? What does the list in admin area show?

How is your webspaces.xml configured?

Hints to debug:
- As you can see in the NewsSitemapProvider.php the Sitemap is created with the help of $entity->getRoutePath(). Please check what you get with this method from your news.

- The first part to the news is controlled by translation ("sulu_news.news"), did you modify this somethere in  your translation?

- When creating/modifying  a news element it should look like this:
![image](https://github.com/manuxi/SuluNewsBundle/assets/11303615/d2a9ed2c-f5b5-47a3-a184-b36d9fc90025)

**Reply by abdellahrk** [reply]
Hello @manuxi, 

> When creating/modifying a news element it should look like this:

 - Yes. it shows that url taken from the title. But the problem is that I cannot modify the `/blog-und-infos` section. Mine shows `news` so, I must use `news` as my default prefix.

> The first part to the news is controlled by translation ("sulu_news.news"), did you modify this somethere in your translation?


- Nope. I don't understand which part you mean here.

**Reply by manuxi** [reply]
'news' should be ok. You can modify the part with the translation string above.

**Reply by abdellahrk** [reply]
I don't understand.

**Reply by manuxi** [reply]
To fit my needs I made it workt to modify the path, which is part of the url.
So, instead of having the path "/news/test-123" you can change it to e.g. "/blog/test-123". 
This is made with the help of a translation, see src/Resources/config/translations/messages.en.yml.
Modify this in your projects translation file (e.g. translations/messages.en.yml) to overwrite. After clearing caches it should work.
But be aware to check the created news items.

**Reply by manuxi** [reply]
Repository
![image](https://github.com/manuxi/SuluNewsBundle/assets/11303615/fc9acca7-ecd6-4deb-b71a-8a96430cd2ed)
Project
![image](https://github.com/manuxi/SuluNewsBundle/assets/11303615/7f2626f7-917a-4080-af6b-cb5e255b3b7f)

**Comment by manuxi** [comment]
Mine looks like this:
`<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<script/>
<sitemap>
<loc>https://[domain]/sitemaps/pages-1.xml</loc>
</sitemap>
<sitemap>
<loc>https://[domain]/sitemaps/events-1.xml</loc>
</sitemap>
<sitemap>
<loc>https://[domain]/sitemaps/news-1.xml</loc>
</sitemap>
<sitemap>
<loc>https://[domain]/sitemaps/abbreviations-1.xml</loc>
</sitemap>
</sitemapindex>`
You can then follow the links as Google would do. Just open e.g. https://[domain]/sitemaps/news-1.xml in your browser

**Reply by abdellahrk** [reply]
Yes. My main sitemap contains `news-1.xml`. Just its content is different.
<img width="755" alt="Screenshot 2024-02-02 at 07 59 02" src="https://github.com/manuxi/SuluNewsBundle/assets/17084652/87f7c461-6da7-4f99-a9c3-e434146ffd4b">

**Comment by manuxi** [comment]
btw. If you like this bundle, feel free to give it a *star*...

**Reply by abdellahrk** [reply]
Yes. I am giving it a start.

**Comment by abdellahrk** [comment]
Can you show me some pointers? I'll like to contribute. 
 I have never created a Symfony Bundle. If you have some resources I can rely on for us to make this package the best. 
Thanks.

**Reply by manuxi** [reply]
Feel free to fork and modify this bundle to fit your needs ;)

**Reply by manuxi** [reply]
Sulu is build on with the help of Symfony, a pretty good php framework.
This shows the basics:
https://symfonycasts.com/tracks/symfony

To dig deeper into Sulu, do the workshop. Sulu is pretty amazing.
https://github.com/sulu/sulu-workshop

If you're working with Symfony (and/or Sulu) I'd recommend using webpack for simpifying js/css handling. symfonycasts.com helps to understand the concepts.

**Comment by abdellahrk** [comment]
Hello @manuxi, no matter what I do, this is how my `news-1.xml` sitemap appears. No errors in the log.


<img width="896" alt="Screenshot 2024-04-23 at 07 06 07" src="https://github.com/manuxi/SuluNewsBundle/assets/17084652/f8f333b3-6dd9-4475-8f75-985d3b027833">

**Reply by manuxi** [reply]
Please ask in Sulu Slack channel.
